src/main/java/by/custom/utilcalculator/domain/tree/CommandTree.java

```diff
@@ -0,0 +1,75 @@
+package by.custom.utilcalculator.domain.tree;
+
+import by.custom.utilcalculator.domain.UserProgress;
+
+import java.util.Objects;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+public class CommandTree {
+
```

Comment:
I believe head of tree should be placed here not in `NodeStorage`

Reply:
NodeStorage has been removed